### ISSUE TITLE
Does AppSyncClient use Amplify Logger?

### ISSUE THREAD

**Issue by hisham** [awslabs/aws-mobile-appsync-sdk-js#355]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
Setting `Amplify.Logger.LOG_LEVEL = 'DEBUG';` does not emit any appsync related logs. 

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

Set Amplify.Logger.LOG_LEVEL = 'DEBUG'; and watch the console logs. 

**What is the expected behavior?**

I should be able to see some appsync related logs. 

**Which versions and which environment (browser, react-native, nodejs) / OS are affected by this issue? Did this work in previous versions?**
Latest versions.

**Comment by greimela** [comment]
I asked myself the same question and figured out the following:
The AppSync client uses https://www.npmjs.com/package/debug for logging.
You can enable the output 
- in nodejs by setting the [correct environment variable](https://www.npmjs.com/package/debug#environment-variables)
- in the browser by adding an [entry to your localStorage](https://www.npmjs.com/package/debug#browser-support). 
If you want to see all logs you can set `localStorage.debug = 'aws-appsync:*'`

**Comment by hisham** [comment]
Cool thanks! I"m using the browser. Tried to set `* localStorage.debug = 'aws-appsync:*'` in the console but nothing new was being logged. Setting 'window.LOG_LEVEL = 'DEBUG'` however in the console enabled all other amplify logging...

**Comment by hisham** [comment]
<img width="669"  src="https://user-images.githubusercontent.com/81853/52359910-0db36680-29f8-11e9-8fcc-d5ce2c73edf7.png">

Still not working for me :S. Attached is screenshot of my localStorage setting in Chrome DevTools. Still no logs appear in the console.

**Comment by greimela** [comment]
Strange.
Are you sure you are using the latest version? Logging has been introduced in `aws-appsync@1.7.0`.

**Comment by hisham** [comment]
Yes I am running 1.7.1. I do see the 'debug' package as a dependency in the aws-appsync package inside my node_modules.

**Comment by manueliglesias** [comment]
@hisham could it be possible that you have some sort of filtering of log messages active?

![image](https://user-images.githubusercontent.com/6154160/53050440-29822800-344e-11e9-8d66-1804cd2e3f51.png)

@greimela instructions are correct

**Comment by hisham** [comment]
Nope nothing outputs and I don't have any filters. You can test the app yourself at https://econsults.seemaeye.com.

**Comment by hisham** [comment]
Actually running in dev mode (localhost) and with aws-appsync:*,* produces debug logs from sockjs-client:main: 

<img width="294" alt="seema institute 2019-02-19 15-40-11" src="https://user-images.githubusercontent.com/81853/53052690-a7e5c680-345c-11e9-9f38-2269c281c3da.png">

But I don't see appsync specific logs. And it seems the debug logs are disabled in production deployed version regardless of local storage setting.

**Comment by manueliglesias** [comment]
Regarding not seeing the messages in prod, I think your bundler might not be including the debug module maybe?



> Actually running in dev mode (localhost) and with aws-appsync:_,_ produces debug logs from sockjs-client:main:

I noticed it is missing the '*', can you try with `aws-appsync:*`

**Comment by isocra** [comment]
How do you enable this in Expo/React Native?

I've tried in my typescript code to put 
```typescript
import { debug } from 'debug';
debug.enable('aws-appsync:*');
```
But this doesn't have any effect.
